src/compressed_tensors/quantization/lifecycle/apply.py

```diff
@@ -78,24 +85,25 @@ def load_pretrained_quantization(model: Module, model_name_or_path: str):
     for name, submodule in iter_named_leaf_modules(model):
         if not is_module_quantized(submodule):
             continue
-        if submodule.quantization_scheme.weights is not None:
-            base_name = "weight"
+        if submodule.quantization_scheme.input_activations is not None:
+            base_name = "input"
             _load_quant_args_from_state_dict(
                 base_name=base_name,
                 module_name=name,
                 module=submodule,
                 state_dict=state_dict,
             )
-        if submodule.quantization_scheme.input_activations is not None:
-            base_name = "input"
+        if submodule.quantization_scheme.output_activations is not None:
+            base_name = "output"
             _load_quant_args_from_state_dict(
                 base_name=base_name,
                 module_name=name,
                 module=submodule,
                 state_dict=state_dict,
             )
-        if submodule.quantization_scheme.output_activations is not None:
-            base_name = "output"
+
+        if load_weight_quantization and submodule.quantization_scheme.weights:
```

Comment:
any reason the order was switched here to do activations first?

Reply:
activations are never compressed in our case (so far) so we can load them right away. 
weight params however are and we want to delay loading them until we need them for decompression, so we're not reloading twice.